### DISCUSSION TITLE
Not all video resolution are shown

### DISCUSSION THREAD

**Discussion by trup** [aclap-dev/video-downloadhelper#687] [Help and Q&A] [answered]
### Your question

Hello.
The latest versions of the program show only 3 video resolution, in fact there are more.
Example https://smotrim.ru/video/2515435

### Addon & CoApp version and details

version: 9.0.1.13
target: mozilla
channel: stable
lang: ru
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\AnnA\\Videos",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 8
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19.

I visited your web page & I got this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/8388ac8f-9810-4f8b-9f74-2ccbdba65ff3)

It's a bit disturbing that VDH didn't manage to actually show the resolutions.  But it did show the bit rates, which is sort of a proxy for the resolution.  You can see that the bit rates are sorted in descending order.

Meanwhile, the Network Monitor did show an HLS master manifest.  That's the one whose Type is vnd.apple.mpegurl.  I ran that through ffprobe & got these results:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/15448758/ffprobe.txt)

There are 5 resolutions listed there & VDH seems to be detecting all 5.  Like I say, it's a bit strange that VDH isn't displaying the values for the resolution but it does appear to be detecting them.  You can see in the right side of the Network Monitor values for BANDWIDTH on each entry.  Those values are reported by ffprobe as variant_bitrate.  Those values kind of correlate with the numbers VDH is reporting as bit rates, but not entirely.  The first 3 bit rates VDH reports do seem to match up with the BANDWIDTH/variant_bitrate values, but the last 2 . . .  I don't know where VDH is getting those numbers.  It doesn't appear to be from the manifest.

In any case, I cannot confirm your observation.  My observation is that VDH is indeed detecting all available resolutions, even if that could be improved.  Perhaps this is an improvement offered by the VDH beta version.  I see you are not using the beta.  Maybe you should.

**Reply by GrampaWildWilly** [reply]
Wait.  I think I know why VDH isn't showing resolutions.  HLS master manifests can include resolution information.  This particular master manifest does not include that.  VDH apparently doesn't go any further than the manifest.  Ffprobe actually reads part of the data from the media stream.  You can see this in the ffprobe log I posted above.  The information about resolution is contained within the media stream itself, even if it isn't in the manifest.  That's why ffprobe could report a resolution for each stream.  VDH apparently uses some other mechanism for detecting media information.  Whatever it's using, it's not getting the resolution.  However, it is getting bit rate information.  That information appears to be different from what ffprobe detects.  Like I said, I don't work for VDH so I'm just guessing based on what you can see here.  Perhaps @paulrouget Paul could explain it better.

**Reply by trup** [reply]
you see 5 video resolutions, and that's correct.
I only have 3.
![res](https://github.com/aclap-dev/video-downloadhelper/assets/3046716/68941ebd-758b-4395-b378-3958cfa02c94)

**Reply by paulrouget** [reply, marked as answer]
In the settings, allow more variants:

<img width="312" alt="Screenshot 2024-05-27 at 15 01 01" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/7e3def0b-7c92-4b94-9e47-2fa6df96d711">

**Reply by trup** [reply]
thanks, it works now. why is this restriction needed?

**Reply by paulrouget** [reply]
Some videos have hundreds of variants.

**Reply by trup** [reply]
Perhaps it’s worth adding a “expand” button instead of/together with the limitation?
